pkg/webhook/manifestwork_webhook.go

```diff
@@ -90,8 +91,13 @@ func (a *ManifestWorkAdmissionHook) validateManifestWorkObj(requestObj runtime.R
 		return fmt.Errorf("manifests should not be empty")
 	}
 
-	if len(work.Spec.Workload.Manifests) > manifestLimit {
-		return fmt.Errorf("number of manifests should not be larger than %d", manifestLimit)
+	totalSize := 0
+	for _, manifest := range work.Spec.Workload.Manifests {
+		totalSize = totalSize + manifest.Size()
+	}
+
+	if totalSize > ManifestLimit {
+		return fmt.Errorf("the size of manifests should not be larger than 50k bytes")
```

Comment:
it is better to also  output the  actual size here.

Reply:
done